ddtrace/tracer/globaltracer.go

```diff
@@ -0,0 +1,89 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"sync/atomic"
+
+	"github.com/DataDog/dd-trace-go/v2/ddtrace"
+)
+
+var (
+	// globalTracer stores the current tracer as *ddtrace.Tracer (pointer to interface). The
+	// atomic.Value type requires types to be consistent, which requires using *ddtrace.Tracer.
+	globalTracer atomic.Value
+)
+
+func init() {
+	var tracer Tracer = &NoopTracer{}
+	globalTracer.Store(&tracer)
+}
+
+// SetGlobalTracer sets the global tracer to t.
+func SetGlobalTracer(t Tracer) {
+	old := *globalTracer.Swap(&t).(*Tracer)
+	if !Testing {
+		old.Stop()
+	}
```

Comment:
This actually confirms what I was saying above... it feels like `Testing` could be passed as a boolean parameter to `SetGlobalTracer` and perhaps renamed to `stopPrevious` or something...